x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmAuthenticateAccessTokenTypeTests.java

```diff
@@ -0,0 +1,167 @@
+        );
+        final JwtIssuerAndRealm jwtIssuerAndRealm = randomJwtIssuerRealmPair();
+        final User user = randomUser(jwtIssuerAndRealm.issuer());
+
+        final SecureString jwt2 = randomJwt(jwtIssuerAndRealm, user);
+        final SecureString clientSecret = JwtRealmInspector.getClientAuthenticationSharedSecret(jwtIssuerAndRealm.realm());
+        doMultipleAuthcAuthzAndVerifySuccess(jwtIssuerAndRealm.realm(), user, jwt2, clientSecret, randomIntBetween(1, 3));
+    }
+
+    @Override
+    protected JwtRealmSettingsBuilder createJwtRealmSettingsBuilder(JwtIssuer jwtIssuer, int authzCount, int jwtCacheSize)
+        throws Exception {
+        final JwtRealmSettingsBuilder jwtRealmSettingsBuilder = super.createJwtRealmSettingsBuilder(jwtIssuer, authzCount, jwtCacheSize);
+        final String realmName = jwtRealmSettingsBuilder.name();
+        final Settings.Builder settingsBuilder = jwtRealmSettingsBuilder.settingsBuilder();
+        settingsBuilder.put(RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.TOKEN_TYPE), "access_token")
+            .putList(
+                RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.ALLOWED_SUBJECTS),
+                jwtIssuer.principals.keySet().stream().toList()
+            );
+
+        if (fallbackSub != null) {
+            settingsBuilder.put(RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.FALLBACK_SUB_CLAIM), fallbackSub);
+        }
+        if (fallbackAud != null) {
+            settingsBuilder.put(RealmSettings.getFullSettingKey(realmName, JwtRealmSettings.FALLBACK_AUD_CLAIM), fallbackAud);
+        }
+
+        return jwtRealmSettingsBuilder;
+    }
+
+    @Override
+    protected SecureString randomJwt(JwtIssuerAndRealm jwtIssuerAndRealm, User user) throws Exception {
+        final JwtIssuer.AlgJwkPair algJwkPair = randomFrom(jwtIssuerAndRealm.issuer().algAndJwksAll);
+        final JWK jwk = algJwkPair.jwk();
+
+        final HashMap<String, Object> otherClaims = new HashMap<>();
+        if (randomBoolean()) {
+            otherClaims.putAll(Map.of("other1", randomAlphaOfLength(10), "other2", randomAlphaOfLength(10)));
+        }
+
+        // Randomly set the fallback claims, it can co-exist with the original one in which case it is ignored
+        String subClaimValue = user.principal();
+        if (fallbackSub != null) {
+            if (randomBoolean()) {
+                // original claim does not exist, so it's the effective fallback
+                otherClaims.put(fallbackSub, subClaimValue);
+                subClaimValue = null;
+            } else {
+                // original claim still exist, in this case, the fallback can be anything and it does not matter
+                otherClaims.put(fallbackSub, randomValueOtherThan(subClaimValue, () -> randomAlphaOfLength(15)));
+            }
+        }
+        // TODO: fallback aud
+        List<String> audClaimValue = JwtRealmInspector.getAllowedAudiences(jwtIssuerAndRealm.realm());
```

Comment:
For now we always set the `aud` claim for access tokens since it is mandated by the `JwtAuthenticationToken`. This will be updated in the next PR.